content/docs/2_cookbook/10_panel/0_multilang-field-options/cookbook-recipe.txt

```diff
@@ -0,0 +1,163 @@
+
+But often, users expect to see the translated option labels when they switch the content language. So, how can we achieve this?
+
+## Options from API
+
+Luckily, field options for the select or multiselect fields can be queried from an API. This API can be a simple route we define in our config. 
+
+Our custom API endpoint will return options for the different languages of our multisite.
+
+```php "/site/config/config.php"
+<?php
+
+return [
+    // other config settings
+    'routes' => [
+        [
+            'pattern' => 'category-options/(:any)',
+            'action' => function($lang) {
+                $categories = [
+                    'en' => [
+                        'architecture' => 'Architecture',
+                        'photography'  => 'Photography',
+                        'design'       => 'Design'
+                    ],
+                    'de' => [
+                        'architecture' => 'Architektur',
+                        'photography'  => 'Fotografie',
+                        'design'       => 'Design'
+                    ],
+                ];
+                return $categories[$lang] ?? [];
+            }
+        ]
+    ],
+];
+````
+
+With this route in place, you can now change the field yaml from our example above like this:
+
+```yaml
+fields:
+  category:
+      label:
+        en: Category
+        de: Kategorie
+      type: select
+      options: api
+      api: "{{kirby.url}}/category-options/{{kirby.language.code}}"
+```
+## Result
+The option labels will now react when you switch the content language:
+
+(columns…)
+
+(image: options-english.png)
+
+++++
+
+(image: options-german.png)
+
+(…columns)
+
+## Using custom language variables
+Instead of hard-coding the translated labels in the array, we can take advantage of Kirby's (link: docs/guide/languages/custom-language-variables text: language variables).
+
+
+
+```php "/site/config/config.php"
+<?php
+
+return [
+    // other config settings
+    'routes' => [
+        [
+            'pattern' => 'category-options/(:any)',
+            'action' => function($lang) {
+                $categories = [
+                    'en' => [
+                        'architecture' => I18n::translate('architecture', null, $lang),
+                        'photography'  => I18n::translate('photography', null, $lang),
+                        'design'       => I18n::translate('design', null, $lang)
+                    ],
+                    'de' => [
+                        'architecture' => I18n::translate('architecture', null, $lang),
+                        'photography'  => I18n::translate('photography', null, $lang),
+                        'design'       => I18n::translate('design', null, $lang)
+                    ],
+                ];
+                return $categories[$lang] ?? [];
```

Comment:
Aren't both language arrays the same now? I guess it can be rewritten to this:

```suggestion
                return [
                    'architecture' => I18n::translate('architecture', null, $lang),
                    'photography'  => I18n::translate('photography', null, $lang),
                    'design'       => I18n::translate('design', null, $lang)
                ];
```

Reply:
You are of course right 🙈 ! And I'm even to stupid to just commit the suggestion 🙈 🙈 🙈 .

Reply:
> needs to be a core feature
Not sure about this, because you  end up with mixed languages in the form.

Reply:
Yes, it's difficult to do in the current blueprint syntax, but having to use the options API is not ideal either.

Reply:
It just dawned on me that this use case can easily be achieved with dynamic blueprints. I think we can throw this recipe away.

Reply:
And I will add this use case to the other recipe instead.